src/SprykerEco/Yves/Payone/Form/DataProvider/BancontactOnlineTransferDataProvider.php

```diff
@@ -36,14 +37,16 @@ public function __construct(PayoneConfig $config)
      */
     public function getData(AbstractTransfer $quoteTransfer): AbstractTransfer
     {
-        /** @var \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer */
-        if ($quoteTransfer->getPayment() !== null) {
+        if ($quoteTransfer instanceof QuoteTransfer && $quoteTransfer->getPayment() !== null) {
             return $quoteTransfer;
         }
 
         $paymentTransfer = new PaymentTransfer();
         $paymentTransfer->setPayone(new PayonePaymentTransfer());
-        $quoteTransfer->setPayment($paymentTransfer);
+
+        if ($quoteTransfer instanceof QuoteTransfer) {
```

Comment:
Why do we need this instanceof check here?

Reply:
It is a condition for passing the validation of PHPStan and code-style sniffer.

Reply:
It was discussed with @AsonUnique.

Reply:
If I remove it I have
`Call to an undefined method Spryker\Shared\Kernel\Transfer\AbstractTransfer::setPayment().`
If I set inline dock block I will have:
`No assignment to $quoteTransfer variable before or after the documentation comment (SlevomatCodingStandard.Commenting.InlineDocCommentDeclaration.NoAssignment)`
because of it we should have this `if` here.

Reply:
OK, but this method can be simplified and the one line moved into the if check.

```suggestion
        if ($quoteTransfer instanceof QuoteTransfer) {
            $quoteTransfer->setPayment((new PaymentTransfer())->setPayone(new PayonePaymentTransfer()));
        }
```

Reply:
Fixed.